src/app/dev/DevToys.Tools/Helpers/LoremIpsum/LipsumGenerator.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Forked from https://github.com/alexcpendleton/NLipsum
+
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace DevToys.Tools.Helpers.LoremIpsum;
+
+/// <summary>
+/// Represents a utility that generates Lipsum from a source.
+/// </summary>
+internal sealed class LipsumGenerator
+{
+    private const string LoremIpsumStartText = "Lorem ipsum dolor sit amet";
+
+    private LipsumsCorpus _lipsum = LipsumsCorpus.LoremIpsum;
+    private StringBuilder _lipsumText = new();
+    private IReadOnlyList<string> _preparedWords = Array.Empty<string>();
+
+    internal LipsumGenerator(LipsumsCorpus lipsum)
+    {
+        _lipsum = lipsum;
+        LipsumText = new StringBuilder(Lipsums.GetText(lipsum));
+    }
+
+    internal LipsumGenerator(string rawData, bool isXml)
+    {
+        /*
+         * If we're receiving an XML string we need to do some
+         * parsing to retrieve the lipsum text.
+         */
+        LipsumText = isXml ?
+            LipsumUtilities.GetTextFromRawXml(rawData) : new StringBuilder(rawData);
+    }
+
+    /// <summary>
+    /// Gets or sets the text used to generate lipsum.
+    /// </summary>
+    internal StringBuilder LipsumText
+    {
+        get => _lipsumText;
+        set
+        {
+            _lipsumText = value;
+            PreparedWords = PrepareWords();
+        }
+    }
+
+    /// <summary>
+    /// Gets the words prepared from the LipsumText.
+    /// </summary>
+    internal IReadOnlyList<string> PreparedWords
+    {
+        get => _preparedWords;
+        private set => _preparedWords = value;
+    }
+
+    /// <summary>
+    /// Generates 'count' features.  The format string will be applied to the feature not the result.
+    /// </summary>
+    /// <param name="count">How many features are desired.</param>
+    /// <param name="feature">The desired feature, such as Paragraph or Sentence.</param>
+    /// <returns></returns>
+    internal string GenerateLipsum(int count, Features feature)
+    {
+        var results = new StringBuilder();
+        string[] data;
+        TextFeature? options = null;
+        if (feature == Features.Paragraphs)
+        {
+            data = GenerateParagraphs(count, FormatStrings.Paragraph.LineBreaks);
+        }
+        else if (feature == Features.Sentences)
+        {
+            data = GenerateSentences(count, FormatStrings.Sentence.Phrase);
+            options = Sentence.Medium;
+        }
+        else if (feature == Features.Words)
+        {
+            data = GenerateWords(count);
+            options = Word.Medium;
+        }
+        else if (feature == Features.Characters)
+        {
+            data = GenerateCharacters(count);
+        }
+        else
+        {
+            throw new NotImplementedException("Sorry, this is not yet implemented.");
```

Comment:
Hello, maybe we should use resx string for this Exception message?

Reply:
The message is not displayed to the user. So let's keep it in English only. Therefore, hardcoded is fine. :-)